File: Nested_Loops_Exercise/train_the_trainers.py
number_of_jury = int(input())
sum_grades = 0
total_grades = 0
counter = 0

name_of_presentation = input()
while True:
    counter += 1
    for i in range(number_of_jury):
        grade = float(input())
        sum_grades += grade
        average_grade = sum_grades / number_of_jury
    print(f'{name_of_presentation} - {average_grade:.2f}.')
    total_grades += average_grade
    sum_grades = 0
    
    name_of_presentation = input()
    if name_of_presentation == "Finish":
        break
        
total_grades /= counter
print(f"Student's final assessment is {total_grades:.2f}.")

File: C++/Basics/NestedLoops_Exercise/TrainTheTrainers/TrainTheTrainers.cpp
#include <iostream>
#include <string>
using namespace std;

int main()
{
	int n; cin >> n;

	cin.ignore();
	cout.setf(ios::fixed);
	cout.precision(2);

	string input;
	getline(cin, input);

	double sumAllGrades = 0;
	int countAllGrades = 0;

	while (input != "Finish")
	{

		double sumGradesForPresentation = 0;
		for (int jury = 1; jury <= n; jury++)
		{
			double grade; cin >> grade;

			sumGradesForPresentation += grade;

			sumAllGrades += grade;
			countAllGrades++;
		}

		double averageGrade = sumGradesForPresentation / n;
		cout << input << " - " << averageGrade << "." << endl;

		cin.ignore();
		getline(cin, input);
	}

	double averageAll = sumAllGrades / countAllGrades;
	cout << "Student's final assessment is " << averageAll << "." << endl;
	
	return 0;
}